src/libs/auth.ts

```diff
@@ -399,40 +415,27 @@ export const isAzureUser = (): boolean => {
   return _.startsWith('https://login.microsoftonline.com', getUser().idp!);
 };
 
-export const processUser = (user: OidcUser | null, isSignInEvent: boolean): void => {
+export const doUserLoaded = (user: OidcUser): void => {
   return authStore.update((state) => {
-    const isSignedIn = !_.isNil(user);
-    const profile: B2cIdTokenClaims | undefined = user?.profile;
-    const userId: string | undefined = profile?.sub;
-    // The following few lines of code are to handle sign-in failures due to privacy tools.
-    if (isSignInEvent && state.isSignedIn === false && !isSignedIn) {
-      // if both of these values are false, it means that the user was initially not signed in (state.isSignedIn === false),
-      // tried to sign in (invoking processUser) and was still not signed in (isSignedIn === false).
-      notify('error', 'Could not sign in', {
-        message: 'Click for more information',
-        detail:
-          'If you are using privacy blockers, they may be preventing you from signing in. Please disable those tools, refresh, and try signing in again.',
-        timeout: 30000,
-      });
-    }
+    const profile: B2cIdTokenClaims = user.profile;
+    // according to IdTokenClaims, this is a mandatory claim and should always exist
+    const userId: string = profile.sub;
     return {
       ...state,
-      isSignedIn,
-      anonymousId: !isSignedIn && state.isSignedIn ? undefined : state.anonymousId,
-      registrationStatus: isSignedIn ? state.registrationStatus : 'uninitialized',
-      termsOfService: initializeTermsOfService(isSignedIn, state),
-      profile: isSignedIn ? state.profile : {},
-      nihStatus: isSignedIn ? state.nihStatus : undefined,
-      fenceStatus: isSignedIn ? state.fenceStatus : {},
+      isSignedIn: true,
+      anonymousId: state.anonymousId,
+      registrationStatus: state.registrationStatus,
+      termsOfService: state.termsOfService,
+      profile: state.profile,
+      nihStatus: state.nihStatus,
+      fenceStatus: state.fenceStatus,
```

Comment:
Are these lines needed anymore? Aren't all these cases where the state is modified covered by `...state`?

Reply:
Removed 👍